Add Formatting Full Messages with I18n section in ActiveRecordValidations guides [ci-skip] (#45470)
rails/rails: guides/source/i18n.md

* `ActiveModel::Errors#generate_message` (which is used by Active Model validations but may also be used manually) uses `model_name.human` and `human_attribute_name` (see above). It also translates the error message and supports translations for inherited class names as explained above in "Error message scopes".

> * `ActiveModel::Errors#full_messages` prepends the attribute name to the error message using a separator that will be looked up from [errors.format](https://github.com/rails/rails/blob/main/activemodel/lib/active_model/locale/en.yml#L4) (and which defaults to `"%{attribute} %{message}"`). To customize `full_messages`, see [`config.active_model.i18n_customize_full_message`][] config.

Reviewer: If we're going to add on to this, I think we should rework it a little:

```suggestion
* `ActiveModel::Errors#full_messages` prepends the attribute name to the error message using a format looked up from `errors.format` (default: [`"%{attribute} %{message}"`](https://github.com/rails/rails/blob/main/activemodel/lib/active_model/locale/en.yml#L4)). To customize the format per model or per attribute, see [`config.active_model.i18n_customize_full_message`][].
```
